### PR TITLE
Extend feature and executor to have an InitializeBeforeBase flag

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

File: src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs [hunk 1]
```diff
@@ -82,38 +82,49 @@ public void Dispose()
...
-                    if (exceptions == null)
-                        exceptions = new List<Exception>();
-
-                    exceptions.Add(ex);
-
-                    this.logger.LogError("An error occurred: '{0}'", ex.ToString());
-
-                    // If we are starting up we need to exit here and stop the node from
-                    // starting up further.
-                    if (!disposing)
```
Comment (dangershony) [inline src/Stratis.Bitcoin/Builder/FullNodeFeatureExecutor.cs]: This is missing
```
  if (!disposing) 
       break; 
```

**Comment by fassadlr** [comment]
Fixes #3021

**Comment by bokobza** [comment]
please check this out:
image

**Comment by fassadlr** [comment]
Tests should now be fixed 👍

**Comment by fassadlr** [comment]
Closing this as I will resubmit PR.

**Pull request closed without merging**